### ISSUE TITLE
[Question] What is filter g?

### ISSUE THREAD

**Issue by bjzhb666** [JngwenYe/LIRF#1]
Thanks for your great work!
In your paper pages 7 and 8,  you mention filters $g$ and $\bar g$.
**What exactly is this filter, is it a network module? How did you get it?**

![image](https://github.com/JngwenYe/LIRF/assets/71651717/1141f259-daa4-460e-9b4b-676dd98ff601)

**Comment by JngwenYe** [comment]
it's just a simple selection operation to calculate the related label possibilities. For example, for fc output: [0.1,0.2,0.2,0.5], if we want to preserve the first two labels, g([0.1,0.2,0.2,0.5]) = [0.1,0.2]

**Comment by bjzhb666** [comment]
Thanks for your quick reply. 
I have another question.
In [https://github.com/JngwenYe/LIRF/blob/main/train_deposit.py#L150-L158 ](https://github.com/JngwenYe/LIRF/blob/main/train_deposit.py#L150-L158), the code does not use `criterionAT(output_batch1, output_teacher_batch1.detach())`, **is the current version of code use our LIRF method?**

**which loss do loss1, loss2 and loss3 correspond to in the paper ($L_{kp}$ etc.)?**

**Comment by bjzhb666** [comment]
And for the return value in [https://github.com/JngwenYe/LIRF/blob/main/train_deposit.py#L174](https://github.com/JngwenYe/LIRF/blob/main/train_deposit.py#L174), are two return values the same?

**Comment by bjzhb666** [comment]
@JngwenYe Can you share the complete code of your work? Thanks

**Comment by XXiaoY** [comment]
> @JngwenYe Can you share the complete code of your work? Thanks
Did you successfully reproduce it

**Comment by bjzhb666** [comment]
> > @JngwenYe Can you share the complete code of your work? Thanks
> > Did you successfully reproduce it

no
